### DISCUSSION TITLE
problem with registry search

### DISCUSSION THREAD

**Discussion by ivansmm** [goharbor/harbor#19724] [Q&A]
Hallo, I'm trying to search harbor registry and it always fails with "unauthorized to list
catalog" error. I am able to push images to repo and I can pull them back. Originally I tried this with harbor running under podman control ()for which I had to do certain configuretion changes), but after it failed I tried with docker-compose and without any intermadiate proxies and it fails with same error also. I also tried it using docker as client (so no podman at all) and still was not able to search the registry. Is there any special setting allowin grgistry serch?
Best regards,
Michael

**Comment by stonezdj** [comment]
What is your search command?

**Comment by ivansmm** [comment]
podman search harbor.host.name/library
It responds with "unauthorized to list catalog"
I am able to push and pull images to/from this /library project as same user.
I am able to list tags for specified image in /library project with skopeo
But serch always returns error

**Comment by ivansmm** [comment]
Just tried with newly released 2.10.0, still same problem.

**Comment by ivansmm** [comment]
I tried with curl and it worked but only when I authenticate as admin user:
```
   curl -u admin:Harbor12345 https://test.intern.local/v2/_catalog
   {"repositories":["ahs/provisioner","ba/nginx", ...., "vga/zookeeper"]}
```
But when I authenticate as same user with docker and run docker search test.intern.local/\<project\>, I get:
```
   Error response from daemon: Unexpected status code 404
```
When I run same command with podman with debug turned on, I'm getting the following output:
```
   DEBU[0000] trying to talk to v1 search endpoint         
   DEBU[0000] GET https://test.intern.local/v2/   
   DEBU[0000] Ping https://test.intern.local/v2/ status 401 
   DEBU[0000] GET https://test.intern.local/v1/search?n=25&q=logit 
   DEBU[0000] error getting search results from v1 endpoint "test.intern.local": invalid status code from registry 404 (Not Found) 
   DEBU[0000] trying to talk to v2 search endpoint         
   DEBU[0000] GET https://test.intern.local/service/token?account=ivans&service=harbor-registry 
   DEBU[0000] GET https://test.intern.local/v2/_catalog 
   ERRO[0000] error getting search results from v2 endpoint "test.intern.local": unauthorized: unauthorized to list catalog: unauthorized to list catalog 
   Error: 1 error occurred:
	* couldn't search registry "test.intern.local": unauthorized: unauthorized to list catalog: unauthorized to list catalog
```
Any hope to get this working?

**Comment by FaraSys** [comment]
I have the same problem.
I use Octupus Deploy and add harbor as External Feed (Docker Registry)
If I create user accout with admin privileges and set this creds in Octopus Deploy search work fine.
If I create robot accout with all 14 privileges and set this robot creds in Octopus search doesnt work.
I get error ``` Authorization to `https://harbor.my.domain/v2/_catalog?n=100` failed. ```

**Comment by hellt** [comment]
Would also +1 this, being able to use `docker/podman search` with harbor would be amazing.

**Comment by groc-soptim** [comment]
(harbor v2.12.0)

Same issue with **oras repo ls**

With non-admin user the command `oras repo ls $REGISTRY/$REPOSITORY` will return a `401 Unauthorized`:

```
Error response from registry: could not list repositories for "harbor.local" with prefix "my/podi/"
unauthorized: unauthorized to list catalog: unauthorized to list catalog

```

Using the admin account when logging in with `oras login $REGISTRY` it works fine

**Comment by dmenneck** [comment]
I'm experiencing the same issue:

```sh
podman search <HARBOR INSTANCE>/
ERRO[0000] error getting search results from v2 endpoint "<HARBOR INSTANCE>": unauthorized: unauthorized to list catalog: unauthorized to list catalog
Error: 1 error occurred:
        * couldn't search registry "<HARBOR INSTANCE>": unauthorized: unauthorized to list catalog: unauthorized to list catalog
```

I can successfully push and pull images.
